packages/data-stores/src/plans/queries/use-plans.ts

```diff
@@ -73,6 +73,7 @@ function usePlans( {
 				} )
 			);
 		},
+		staleTime: 3600000, // 1 hour in ms
```

Comment:
Why is this changed needed? Does this mean that if we ship some API changes to update the pricing or the name of some plans, they will not be visible for 1 hour? Because if that's the case, I think we need to discuss this change more deeply with other teams. With the current code, the expectation is that API changes to the plans will be reflected immediately in Calypso.

Reply:
Bogdan mentioned that there were refetches when lazy loading a new batch of themes:

> We seem to make a new /plans request on each theme card display (on mobile) - this is more serious since it's happening on Logged out too (returning a lot of 403 error responses)

After some investigation I founds that it happens because the `ThemeTierUpgradeBadge` triggers a `usePlans` call when it's rendered, so that's what I set a `staleTime` of one hour.

> Does this mean that if we ship some API changes to update the pricing or the name of some plans, they will not be applied for up to 1 hour?

No, they would be applied immediately, but users that are already navigating the site without actually reloading it would not see the new data until one hour has gone by after the last hook call.

Anyway, the value can be set to one minute as well, I don't think a request per minute while the user scrolls the themes is something to be concerned. Wdyt?

Reply:
> After some investigation I founds that it happens because the `ThemeTierUpgradeBadge` triggers a `usePlans` call when it's rendered, so that's what I set a `staleTime` of one hour.

This made me think 🤔 It seems that we're only interested in lazy loading the screenshots, right? So can we then move the lazy loading logic from the whole theme card to only the screenshots? That way the `ThemeTierUpgradeBadge` component (and the `usePlans` hook) won't be affected by these changes. 

I also wonder, why these changes increase the number of requests that `ThemeTierUpgradeBadge` component make to `/plans` from one per page load to one per each theme card? That might be a symptom of an underlying issue that we should fix (instead of masking it with a `staleTime`).

Reply:
> So can we then move the lazy loading logic from the whole theme card to only the screenshots? That way the `ThemeTierUpgradeBadge` component (and the `usePlans` hook) won't be affected by these changes.

I thank that would still mean replacing the whole content rendered by `ThemeCard` with what's on the `renderPlaceholder` function of the `Theme` component. So the `badge` would be still re-rendered and triggering the `usePlans` hook.



> I also wonder, why these changes increase the number of requests that `ThemeTierUpgradeBadge` component make to `/plans` from one per page load to one per each theme card?

As I understand it, it's the expected `useQuery` behavior. Having the default `stleTime: 0` means that react query will always trigger a background request to the plans endpoint when a new instance of the `useQuery` hook is rendered, since the cached data is considered stale. That doesn't happen currently on production because all the themes are rendered at the same time and so at no point there's a new `usePlans` instance.

Reply:
> I thank that would still mean replacing the whole content rendered by `ThemeCard` with what's on the `renderPlaceholder` function of the Theme component.

I meant the specific component used for the screenshot returned by the `renderScreenshot` function. But that made wonder: why are over-complicating this with an `InView` component when we already have a native solution with the `loading="lazy"` attribute that makes all the work? So I opened an alternative solution: https://github.com/Automattic/wp-calypso/pull/96961